### ISSUE TITLE
Add the ability to specify a size when launching a session

### ISSUE THREAD

**Issue by vlj91** [alces-software/clusterware#85]
As above - it would be useful if we could specify a size when launching a session

**Comment by bobwhitelock** [comment]
How would we like this to be specified:
- as a positional parameter to the command e.g. `alces session start gnome --geometry 1920x1080`? 
- as a config value in `etc/session.rc`?
- both of these with the former overriding the latter?
- some other way?

**Comment by mjtko** [comment]
Yes! :)

I would have `alces session start --geometry <geom> gnome` rather than the other way round.

A config value in `etc/session.rc` would be good too!

The former should override the latter.

I'm open to suggestions for other ways, but I think you have them covered!

**Comment by bobwhitelock** [comment]
OK! Your suggestion would be consistent with the other optional parameters so that makes sense. Though as an aside, is there a reason (other than it not being a priority to add) that we only support optional parameters that way round rather than both? Typically I would expect both to work e.g. `git commit -m stuff a/file` and `git commit a/file -m stuff` are both valid. May be worth adding at some point for usability?

Regarding this issue again, only other place I would think you could configure things would be a user-specific `~/.clusterware/sessions.rc` file, but I didn't think we support any user-specific config (yet), so this seems unnecessary.

**Comment by mjtko** [comment]
Reason: nope. :) Happy for that to be added!  There is an option parser available in bash: http://wiki.bash-hackers.org/howto/getopts_tutorial might help. Probably worth opening another issue to cover that and attacking all the commands against that one?

I like the idea of a user-specific file.  I recently introduced a general Clusterware settings file in `~/.config/clusterware/settings.rc`, so it might be worth adding an optional `sessions.rc` in the same place.  Up to you though, the main requirement here is allowing the session size to be specified at launch time, so that's the priority.  :)

**Comment by bobwhitelock** [comment]
Ok - I'll stick with the main issue for the moment and can come back to more option parsing/ user configs when/if that becomes a priority.

**Comment by bobwhitelock** [comment]
@mjtko - I tried to enable all base session types to test things worked correctly, I was able to enable them all apart from trinity which I've had some issues with, wondered if you know why.

When I first tried enabling it I got this:

```
[vagrant@login1(el7cluster) ~]$ alces session enable trinity
  >>> Loaded plugins: fastestmirror
  >>> Cannot open: http://mirror.ppa.trinitydesktop.org/trinity/trinity/rpm/el7/trinity-r14/RPMS/noarch/trinity-repo-14.0.2-1.el7.noarch.rpm. Skipping.
  >>> Error: Nothing to do
  >>> Loaded plugins: fastestmirror
  >>> Examining /var/tmp/yum-root-Bd6_dr/libmad-0.15.1b-4.el7.x86_64.rpm: libmad-0.15.1b-4.el7.x86_64
  >>> Marking /var/tmp/yum-root-Bd6_dr/libmad-0.15.1b-4.el7.x86_64.rpm to be installed
  >>> Resolving Dependencies
  >>> --> Running transaction check
  >>> ---> Package libmad.x86_64 0:0.15.1b-4.el7 will be installed
  >>> --> Processing Dependency: libmad0 = 0.15.1b-4.el7 for package: libmad-0.15.1b-4.el7.x86_64
  >>> Loading mirror speeds from cached hostfile
  >>>  * base: mirrors.coreix.net
  >>>  * extras: mirrors.vooservers.com
  >>>  * updates: mirrors.vooservers.com
  >>> --> Finished Dependency Resolution
  >>> Error: Package: libmad-0.15.1b-4.el7.x86_64 (/libmad-0.15.1b-4.el7.x86_64)
  >>>            Requires: libmad0 = 0.15.1b-4.el7
  >>>  You could try using --skip-broken to work around the problem
  >>>  You could try running: rpm -Va --nofiles --nodigest
  >>> Loaded plugins: fastestmirror
  >>> Loading mirror speeds from cached hostfile
  >>>  * base: mirrors.coreix.net
  >>>  * extras: mirrors.vooservers.com
  >>>  * updates: mirrors.vooservers.com
  >>> No package trinity-tdebase available.
  >>> Error: Nothing to do
alces session enable: unable to enable session: base/trinity
```

I tried this a couple of times in case it was a fluke then noticed the mirror seemed unavailable but `http://mirror.ppa.trinitydesktop.org/trinity/trinity/rpm/el7/trinity-r14/RPMS/noarch/trinity-repo-14.0.3-1.el7.noarch.rpm` was (seems the version has incremented), so I changed the line in `trinity/metadata.yml` to this.

I then get:

```
[vagrant@login1(el7cluster) ~]$ alces session enable trinity
  >>> This session cannot be enabled as Alces Clusterware v1.4.1 is too old.
  >>> Please upgrade to Alces Clusterware v1.5.0 or higher.
alces session enable: unable to enable session: base/trinity
```

I assume this is not actually true since I'm branched off `develop`, so changed the version in `/opt/clusterware/etc/clusterware.rc` to `1.5.0`.

However I then get:

```
[vagrant@login1(el7cluster) ~]$ alces session enable trinity 
  >>> Sorry, the Trinity repository is prohibited.
```

Various other session types also won't enable (I created a fresh VM just before running the first commands in this comment to avoid issues caused by some other change), for example:

```
[vagrant@login1(el7cluster) ~]$ alces session enable icewm                                                                               
  >>> Unknown repository: epel
  >>> Sorry, the EPEL repository is not available.
alces session enable: unable to enable session: base/icewm
[vagrant@login1(el7cluster) ~]$ 
[vagrant@login1(el7cluster) ~]$ alces session enable xfce                                                                                
  >>> Unknown repository: epel
  >>> Sorry, the EPEL repository is not available.
alces session enable: unable to enable session: base/xfce
[vagrant@login1(el7cluster) ~]$ 
[vagrant@login1(el7cluster) ~]$ alces session enable chrome                                                                              
  >>> Sorry, the Google Chrome repository is prohibited.
alces session enable: unable to enable session: base/chrome
```

Any ideas?

**Comment by mjtko** [comment]
My theory is that `cw_DIST` is set in the shell, so `distro.rc` isn't being sourced by the `distro.functions.sh` library:

https://github.com/alces-software/clusterware/blob/develop/lib/functions/distro.functions.sh#L22

I'll take a closer look at the distro repo code and get back to you!

**Comment by bobwhitelock** [comment]
`cw_DIST` is set to `el7` in my VM after a new login (at least now, not sure if that's changed somehow since the VM was brand new). Should it not be?

**Comment by mjtko** [comment]
I think at one point it wasn't exported, but has been so for a while (possibly since after I implemented the distro repo stuff).  I'll have a look and see what it effects to not do that (possibly a lot) and/or change the loading logic in `etc/distro.rc`.

**Comment by mjtko** [comment]
Ok, looks like this is working as expected -- think the problems you had (other than the Trinity issue now open as alces-software/clusterware-sessions#8) were based on the fact you'd built from master (`1.4.1`) initially.

**Comment by bobwhitelock** [comment]
I think you're right, now that the VM was built from `develop` that seems to be the real issue.
